code/modules/surgery/organs/tongue.dm

```diff
@@ -496,18 +505,110 @@ GLOBAL_LIST_INIT(english_to_zombie, list())
+		return COMPONENT_CANNOT_SPEAK
+
+	switch(check_signables_state())
+		if(SIGN_HANDS_FULL) // Full hands
+			source.visible_message("tries to sign, but can't with [source.p_their()] hands full!", visible_message_flags = EMOTE_MESSAGE)
+			return COMPONENT_CANNOT_SPEAK
+
+		if(SIGN_CUFFED) // Restrained
+			source.visible_message("tries to sign, but can't with [source.p_their()] hands bound!", visible_message_flags = EMOTE_MESSAGE)
+			return COMPONENT_CANNOT_SPEAK
+
+		if(SIGN_ARMLESS) // No arms
+			to_chat(source, span_warning("You can't sign with no hands!"))
+			return COMPONENT_CANNOT_SPEAK
+
+		if(SIGN_ARMS_DISABLED) // Arms but they're disabled
+			to_chat(source, span_warning("Your can't sign with your hands right now!"))
+			return COMPONENT_CANNOT_SPEAK
+
+		if(SIGN_TRAIT_BLOCKED) // Hands blocked or emote mute
+			to_chat(source, span_warning("You can't sign at the moment!"))
+			return COMPONENT_CANNOT_SPEAK
+
+	// Assuming none of the above fail, sign language users can speak
+	// regardless of being muzzled or mute toxin'd or whatever.
+	return COMPONENT_CAN_ALWAYS_SPEAK
 
+/// Signal proc for [COMSIG_LIVING_TREAT_MESSAGE] that stars out our message if we only have 1 hand free
+/obj/item/organ/internal/tongue/tied/proc/on_treat_message(mob/living/source, list/message_args)
+	SIGNAL_HANDLER
+
+	if(check_signables_state() == SIGN_ONE_HAND)
+		message_args[TREAT_MESSAGE_MESSAGE] = stars(message_args[TREAT_MESSAGE_MESSAGE])
+
+/// Signal proc for [COMSIG_MOVABLE_USING_RADIO] that disallows us from speaking on comms if we don't have the special trait
+/// Being unable to sign, or having our message be starred out, is handled by the above two signal procs.
+/obj/item/organ/internal/tongue/tied/proc/on_use_radio(atom/movable/source, obj/item/radio/radio)
+	SIGNAL_HANDLER
+
+	return HAS_TRAIT(source, TRAIT_CAN_SIGN_ON_COMMS) ? NONE : COMPONENT_CANNOT_USE_RADIO
+
+/// Checks to see what state this person is in and if they are able to sign or not.
+/obj/item/organ/internal/tongue/tied/proc/check_signables_state()
+	if(!owner)
+		CRASH("[type] called check_signables_state without an owner.")
+
+	// See how many hands we can actually use (this counts disabled / missing limbs for us)
+	var/total_hands = owner.usable_hands
+	// Look ma, no hands!
+	if(total_hands <= 0)
+		// Either our hands are still attached (just disabled) or they're gone entirely
+		return owner.num_hands > 0 ? SIGN_ARMS_DISABLED : SIGN_ARMLESS
+
+	// Now let's see how many of our hands is holding something
+	var/busy_hands = 0
+	// Yes held_items can contain null values, which represents empty hands,
+	// I'm just saving myself a variable cast by using as anything
+	for(var/obj/item/held_item as anything in owner.held_items)
+		// items like slappers/zombie claws/etc. should be ignored
+		if(isnull(held_item) || held_item.item_flags & HAND_ITEM)
+			continue
+
+		busy_hands++
+
+	// Handcuffed or otherwise restrained - can't talk
+	if(HAS_TRAIT(owner, TRAIT_RESTRAINED))
```

Comment:
Handcuffs are for two hands only.  If one arm gets dismembered then handcuffs can't be applied.  Also if the arm gets chopped off, the cuffs are removed.  

https://github.com/tgstation/tgstation/blob/a2a2c2da24293785ab4ddc52bd7ec248d0dbcf53/code/modules/surgery/bodyparts/dismemberment.dm#L240-L249

With that consistency in mind, if someone has 3 arms, they should still have a free arm that allows them to sign.  The code for multihand use still exists in our codebase and can only be triggered with admin powers right now.  Apparently some downstreams still use this code as a feature - https://github.com/tgstation/tgstation/pull/69840#issuecomment-1257090020

I'm not a big fan of adding support for features that are deprecated.  It seems silly to me, but it was specifically requested so I obliged.

Reply:
Other than this nitpick, the rest of the PR looks great.  You're refactor of my sign language code is better than what I had before.  So +1 for that.

Reply:
It's not deprecated, please stop saying it is. It's unused, but it's not marked for removal, and can be fixed up with not a terrible amount of effort.

Reply:
The PR that added multihand _never_ actually included a way for the feature to be obtainable in game @Kapu1178 
- #20283

> Even Willard's suggestion in that PR didn't account for multi-hand+handcuffed, it early returned "if cuffed at all"

From my understanding it was pseudocode (since it broke when I tested it) but @JohnFulpWillard can clarify.

Both handsigning and handcuffs have functionality that is affected when a person has one hand vs two. This functionality overlaps since handsigning takes into account handcuffs.  I was specifically asked to add +3 hand support for handsigning so that's exactly what I did.  Nowhere in the codebase to my knowledge directly supports features involving +3 hands.  Downstreams may use it, but using that as justification is irrelevant.

> If we actually WANT to care about multi-hand+handcuffed, it should be supported by the code - which it is not currently. 

Agree 100%.  It would require a large amount of support to account for +3 hands all across the codebase, not to mention balance issues.  Adding support _prematurely_ is a gamble that later on down the road someone will use it.

---

My thought process was basically this:

- Fix sign langauge checks with tongue tied
- Asked by maintainer to add multihand support just in case...
- Add support for multihand shenanigans
- PR gets merged
- 2nd maintainer removes the multihand support from the code that I _just_ wrote
- ... (awkward silence) ...

Reply:
Necro'ing this but #45983 is a bug report I just came across dealing with multihands.  I'm marking it as `Not a bug` for now.